Source/WTF/wtf/CompactVariantOperations.h

```diff
@@ -78,7 +85,7 @@ template<CompactVariantAlternative... Ts> struct CompactVariantOperations {
 
     static constexpr Index decodedIndex(Storage value)
     {
-        return static_cast<Index>(static_cast<uint8_t>(value >> indexShift));
+        return static_cast<Index>(static_cast<uint8_t>((value & indexMask) >> indexShift));
```

Comment:
Seems unnecessary to mask before shifting. I guess if we want to use `indexMask` or `payloadMask` here we could stick with doing it before shifting. Could shift first and mask with `0xFF`, which is `unshiftedIndexMask` (or we could call it `indexMask` and call the other one `shiftedIndexMask`).

Reply:
I don't want to introduce more constants so let's keep this.